mozmeao/basket
# Move to pip-compile-multi

This changeset swaps Basket form pip-tools to using pip-compile-multi as our chosen dependency helper, plus retains and evolves a Makefile-driven approach to using pip-compile-multi.

Notes: 

* UPDATED: this continues to compile requirements inside the Docker container

* for now, while we generate separate `requirements/prod.txt` and `requirements/dev.txt` files, we continue to use the superset of dependencies (requirements/dev.txt) for building the single Docker image. Splitting Docker images is deferred to #784 to keep this change cleaner.

* one dependency - the FuelSDK for Salesforce - had to be removed as it has a very stale dependency and patching the supply chain isn't worth it as FuelSDK isn't used in the Basket codebase any more, it appears.

### To test
* Run `make compile-requirements` 
* Run `make compile-requirements` again and confirm that the deps are rebuilt, but have no changes
* Run `make clean build test` to ensure that full images still build
* Check CI is happy
